src/index.ts

```diff
@@ -515,6 +545,26 @@ export function create(rawOptions: CreateOptions = {}): Service {
     ...(rawOptions.require || []),
   ];
 
+  // Experimental REPL await is not compatible targets lower than ES2018
+  const targetSupportsTla = config.options.target! >= ScriptTarget.ES2018;
+  if (options.experimentalReplAwait === true && !targetSupportsTla) {
+    throw new Error(
+      'Experimental REPL await is not compatible with targets lower than ES2018'
+    );
+  }
+  // Top-level await was added in TS 3.8
+  const tsVersionSupportsTla = versionGte(ts.version, '3.8.0');
+  if (options.experimentalReplAwait === true && !tsVersionSupportsTla) {
```

Comment:
Even if it's not supported in TS < 3.8, it still can be used in `transpileOnly`, so I'd suggest to expand the conditional and only abort if `transpileOnly !== true`

Reply:
I think the problem is that the emit may not be correct in all situations, and I'd rather not support this use-case until someone demonstrates that it's very valuable.  If someone *really* needs REPL top-level await, I'm betting they can use TS 3.8 or newer.

Reply:
https://www.typescriptlang.org/play?target=5&module=1&ts=3.7.5#code/IYd2EsBcAIG8A8Bc0CMBfA3AWAFAFN4AHAewCcZZMg

Reply:
Sure, let's keep as it then.